Facing errors while running test cases

### FerretDB version

Latest nightly

### Backend

PostgresSQL

### Environment

- OS: MAC OS Monterey
- Deployment:
- Deployment details:


### What did you do?

I was setting up the local environment using contribute guidlines. Here is the process I followed

1. task init
2. task env-pull
3. task env-up
4. task test

### What did you expect to see?

No error messages while running test cases.

### What did you see instead?

Here is the error message

```
ferretdb_postgres          | 2023-06-26 16:01:17.117 NDT [1013] ERROR:  schema "TestUpdateArrayCompatPull-Array_pg" does not exist
ferretdb_postgres          | 2023-06-26 16:01:17.117 NDT [1013] STATEMENT:  DROP SCHEMA "TestUpdateArrayCompatPull-Array_pg" CASCADE
ferretdb_postgres          | 2023-06-26 16:01:17.290 NDT [1017] ERROR:  schema "TestDeleteCompat-Two_pg" does not exist
ferretdb_postgres          | 2023-06-26 16:01:17.290 NDT [1017] STATEMENT:  DROP SCHEMA "TestDeleteCompat-Two_pg" CASCADE
ferretdb_postgres          | 2023-06-26 16:01:18.462 NDT [1021] ERROR:  schema "TestDeleteCompat-UnorderedError_pg" does not exist
ferretdb_postgres          | 2023-06-26 16:01:18.462 NDT [1021] STATEMENT:  DROP SCHEMA "TestDeleteCompat-UnorderedError_pg" CASCADE
ferretdb_postgres          | 2023-06-26 16:01:18.606 NDT [1025] ERROR:  schema "TestDeleteCompat-One_pg" does not exist
ferretdb_postgres          | 2023-06-26 16:01:18.606 NDT [1025] STATEMENT:  DROP SCHEMA "TestDeleteCompat-One_pg" CASCADE
```

```
wire_test.go:136: 
        	Error Trace:	/Users/kiran/projects/FerretDB/internal/wire/wire_test.go:136
        	            				/Users/kiran/projects/FerretDB/internal/wire/op_query_test.go:109
        	            				/usr/local/go/src/testing/fuzz.go:722
        	            				/usr/local/go/src/runtime/asm_arm64.s:1172
        	Error:      	Received unexpected error:
        	            	[records_test.go:88 wire.loadRecords] ../../tmp/records/00/00e58bd132ce4dd24875fac15b7b56635bb596f7ed50cd4454433ed3c2f46c41.bin: [msg_body.go:78 wire.ReadMessage] [op_msg.go:225 wire.(*OpMsg).UnmarshalBinary] wire.OpMsg.Document: validation failed for { debugError: { NaN: nan.0 }, $db: "TestDebugError" } with: NaN is not supported
        	Test:       	FuzzQuery
--- FAIL: FuzzReply (5.07s)
    wire_test.go:136: 
        	Error Trace:	/Users/kiran/projects/FerretDB/internal/wire/wire_test.go:136
        	            				/Users/kiran/projects/FerretDB/internal/wire/op_reply_test.go:100
        	            				/usr/local/go/src/testing/fuzz.go:722
        	            				/usr/local/go/src/runtime/asm_arm64.s:1172
        	Error:      	Received unexpected error:
        	            	[records_test.go:88 wire.loadRecords] ../../tmp/records/00/00e58bd132ce4dd24875fac15b7b56635bb596f7ed50cd4454433ed3c2f46c41.bin: [msg_body.go:78 wire.ReadMessage] [op_msg.go:225 wire.(*OpMsg).UnmarshalBinary] wire.OpMsg.Document: validation failed for { debugError: { NaN: nan.0 }, $db: "TestDebugError" } with: NaN is not supported
        	Test:       	FuzzReply
FAIL
```

But I was able to run task env-data and then able to list all collections using show collections command from mongosh.


![](https://user-images.githubusercontent.com/17452039/248905232-0a2e9f0a-6aa7-4fc1-a52f-949a5709170a.png)




The file [00e58bd132ce4dd24875fac15b7b56635bb596f7ed50cd4454433ed3c2f46c41.bin](https://drive.google.com/file/d/1LnOagRHX8Fg-fWvx3hQxS_T0KSNxb6bx/view?usp=sharing)

--- a/internal/handlers/sjson/sjson_test.go
+++ b/internal/handlers/sjson/sjson_test.go
@@ -169,6 +169,10 @@ func fuzzJSON(f *testing.F, testCases []testCase, newFunc func() sjsontype) {
 		var n int
 
 		for _, rec := range records {
+			if rec.Body == nil {
+				continue
+			}
+
 			var docs []*types.Document
 
 			switch b := rec.Body.(type) {

--- a/internal/wire/record.go
+++ b/internal/wire/record.go
@@ -27,8 +27,11 @@ import (
 
 // Record represents a single recorded wire protocol message, loaded from a .bin file.
 type Record struct {
-	Header  *MsgHeader
-	Body    MsgBody
+	// those may be unset if message is invalid
+	Header *MsgHeader
+	Body   MsgBody
+
+	// those are always set
 	HeaderB []byte
 	BodyB   []byte
 }
@@ -96,8 +99,10 @@ func loadRecordFile(file string) ([]Record, error) {
 			break
 		}
 
+		// TODO we should still set HeaderB and BodyB
+		// https://github.com/FerretDB/FerretDB/issues/1636
 		if err != nil {
-			return nil, lazyerrors.Error(err)
+			break
 		}
 
 		headerB, err := header.MarshalBinary()

--- a/internal/wire/wire_test.go
+++ b/internal/wire/wire_test.go
@@ -146,6 +146,10 @@ func fuzzMessages(f *testing.F, testCases []testCase) {
 		require.NoError(f, err)
 
 		for _, rec := range records {
+			if rec.HeaderB == nil || rec.BodyB == nil {
+				continue
+			}
+
 			b := make([]byte, 0, len(rec.HeaderB)+len(rec.BodyB))
 			b = append(b, rec.HeaderB...)
 			b = append(b, rec.BodyB...)

